code/wdb/datastripe.q

```diff
@@ -45,30 +76,29 @@ upserttopartition:{[dir;tablename;keycol;enumdata;nextp]
 	  (directory;enumdata);
 	  {[e] .lg.e[`upserttopartition;"Failed to save table to disk : ",e];'e}
 	];
-	};	
+	};
 
 savetablesoverperiod:{[dir;tablename;nextp]
-	/- function to get keycol for table from access table
-	keycol:`sym; / variable keycol functionality added in TCS-73
-	/- get distinct values to partition table on
-	partitionlist:raze value each ?[[`.]tablename;();1b;enlist[keycol]!enlist keycol];
-	/- enumerate table to be upserted and get each table by sym
-	enumdata:{[dir;tablename;keycol;nextp;s] .Q.en[dir;0!?[[`.]tablename;((<;`time;nextp);(=;keycol;enlist s));0b;()]]}[dir;tablename;keycol;nextp]'[partitionlist];
-	/-upsert table to partition
-	upserttopartition[dir;tablename;keycol;;nextp] each enumdata;
-	/- delete data from last period
-	.[.ds.deletetablebefore;(tablename;`time;nextp)];
-	/ .[{![x;enlist(<;`time;y);0b;0#`]};(tablename;nextp)];
-	/- run a garbage collection (if enabled)
-	.gc.run[];
-	};
+    /- function to get keycol for table from access table
+    keycol:$[.wdb.tablekeycols[tablename]=`;`sym;.wdb.tablekeycols[tablename]];
+    /- get distint values to partition table on
+    partitionlist:raze value each ?[[`.]tablename;();1b;enlist[keycol]!enlist keycol];
+    /- enumerate table to be upserted and get each table by sym
+    enumdata:{[dir;tablename;keycol;nextp;s] .Q.en[dir;0!?[[`.]tablename;((<;`time;nextp);(=;keycol;enlist s));0b;()]]}[dir;tablename;keycol;nextp]'[partitionlist];
+    /-upsert table to partition
+    upserttopartition[dir;tablename;keycol;;nextp] each enumdata;
```

Comment:
I might be missing something here, but will this upsert each of the split tables to same partition? I don't see `partitionlist` being used here, so it seems like we just split it & then we're upserting it all to same partition - in which case, why do we need to split it?

Reply:
This is handled in upsert to partition function, function has a line to take each unique value from the keycolumn and upsert to unique partition

Reply:
OK - so why do we need to split it before sending to that function then? (Could be a valid reason for this that I'm missing)

Reply:
Still looking over this but upserttopartition takes each split table, determines which syms it contains and then upserts it to the correct partition.

Reply:
The upserttopartition function was recalculating the partitionlist and then upserting to that partition, by taking the partitionlist as a variable in the upserttopartition function we can remove this redundant code and hopefully make this more clear.